### ISSUE TITLE
Update call failed: the desired mutation for the following field(s) is invalid: [nodeConfig.0.OauthScopes.#...

### ISSUE THREAD

**Issue by Bobgy** [GoogleCloudPlatform/k8s-config-connector#259]
**Describe the bug**
When deploying Kubeflow on GCP: https://github.com/kubeflow/gcp-blueprints.
One user reported a problem of getting
```
      message: 'Update call failed: the desired mutation for the following field(s)
        is invalid: [nodeConfig.0.OauthScopes.# ipAllocationPolicy.# initialNodeCount
        nodeConfig.0.OauthScopes.3859019814 nodeConfig.0.MachineType nodeConfig.0.ServiceAccount]'
      reason: UpdateFailed
```
for a containercluster resource.
Ref: https://github.com/kubeflow/kubeflow/issues/5223#issuecomment-679242861

Even though we asked them to delete the resource and recreate one: https://github.com/kubeflow/kubeflow/issues/5223#issuecomment-679198124. Therefore this seems like a bug.

**ConfigConnector Version**
1.7.1

**To Reproduce**
*Steps to reproduce the behavior:*
not reproducible yet

*YAML snippets:*

```yaml
apiVersion: v1
items:
- apiVersion: container.cnrm.cloud.google.com/v1beta1
  kind: ContainerCluster
  metadata:
    annotations:
      cnrm.cloud.google.com/management-conflict-prevention-policy: resource
      cnrm.cloud.google.com/project-id: XXXX
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"container.cnrm.cloud.google.com/v1beta1","kind":"ContainerCluster","metadata":{"annotations":{},"clusterName":"XXXX/us-east1-b/xx","labels":{"kf-name":"xxf","mesh_id":"XXXX_us-east1-b_xx"},"name":"xx","namespace":"XXXX"},"spec":{"clusterAutoscaling":{"autoProvisioningDefaults":{"oauthScopes":["https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/monitoring","https://www.googleapis.com/auth/devstorage.read_only"],"serviceAccountRef":{"name":"xx-vm"}},"enabled":true,"resourceLimits":[{"maximum":128,"resourceType":"cpu"},{"maximum":2000,"resourceType":"memory"},{"maximum":16,"resourceType":"nvidia-tesla-k80"}]},"initialNodeCount":2,"location":"us-east1-b","loggingService":"logging.googleapis.com/kubernetes","monitoringService":"monitoring.googleapis.com/kubernetes","nodeConfig":{"machineType":"n1-standard-8","metadata":{"disable-legacy-endpoints":"true"},"oauthScopes":["https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/monitoring","https://www.googleapis.com/auth/devstorage.read_only"],"serviceAccountRef":{"name":"xx-vm"},"workloadMetadataConfig":{"nodeMetadata":"GKE_METADATA_SERVER"}},"releaseChannel":{"channel":"REGULAR"},"workloadIdentityConfig":{"identityNamespace":"XXXX.svc.id.goog"}}}
    creationTimestamp: "2020-08-24T16:00:09Z"
    generation: 20
    labels:
      kf-name: xx
      mesh_id: XXXX_us-east1-b_xx
    name: xx
    namespace: XXXX
    resourceVersion: "1290712"
    selfLink: /apis/container.cnrm.cloud.google.com/v1beta1/namespaces/XXXX/containerclusters/xx
    uid: 0c9868a3-d248-430a-aa42-7543fc2b6b09
  spec:
    clusterAutoscaling:
      autoProvisioningDefaults:
        oauthScopes:
        - https://www.googleapis.com/auth/logging.write
        - https://www.googleapis.com/auth/monitoring
        - https://www.googleapis.com/auth/devstorage.read_only
        serviceAccountRef:
          name: xx-vm
      enabled: true
      resourceLimits:
      - maximum: 128
        resourceType: cpu
      - maximum: 2000
        resourceType: memory
      - maximum: 16
        resourceType: nvidia-tesla-k80
    initialNodeCount: 2
    location: us-east1-b
    loggingService: logging.googleapis.com/kubernetes
    monitoringService: monitoring.googleapis.com/kubernetes
    nodeConfig:
      machineType: n1-standard-8
      metadata:
        disable-legacy-endpoints: "true"
      oauthScopes:
      - https://www.googleapis.com/auth/logging.write
      - https://www.googleapis.com/auth/monitoring
      - https://www.googleapis.com/auth/devstorage.read_only
      serviceAccountRef:
        name: xx-vm
      workloadMetadataConfig:
        nodeMetadata: GKE_METADATA_SERVER
    releaseChannel:
      channel: REGULAR
    workloadIdentityConfig:
      identityNamespace: XXXX.svc.id.goog
  status:
    conditions:
    - lastTransitionTime: "2020-08-24T16:00:12Z"
      message: 'Update call failed: the desired mutation for the following field(s)
        is invalid: [nodeConfig.0.OauthScopes.# ipAllocationPolicy.# initialNodeCount
        nodeConfig.0.OauthScopes.3859019814 nodeConfig.0.MachineType nodeConfig.0.ServiceAccount]'
      reason: UpdateFailed
      status: "False"
      type: Ready
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Comment by cm185255** [comment]
@Bobgy thanks for opening this ticket. I tried what you mentioned on #5223 
```
kubectl --context=${MGMT-CTXT} --namespace ${PROJECT} delete containercluster ${NAM}
# wait a while
gcloud --project=${PROJECT} container clusters list
# verify the cluster has already been deleted
make apply
```
but could not complete the steps because `gcloud --project=${PROJECT} container clusters list` always returned 
```
NAME       LOCATION    MASTER_VERSION  MASTER_IP      MACHINE_TYPE   NODE_VERSION   NUM_NODES  STATUS
xx        us-east1-b  1.16.13-gke.1   35.237.216.43  n1-standard-4  1.16.13-gke.1  3          RUNNING
```
However, when I re-ran `kubectl --context=${MGMT-CTXT} --namespace ${PROJECT} delete containercluster ${NAM}` it returned 
```
Error from server (NotFound): containerclusters.container.cnrm.cloud.google.com "dip-ai-kf" not found
```
as though the cluster *was* correctly deleted. I'm not sure what to make of this... I will give it some more time to see if  `gcloud --project=${PROJECT} container clusters list` will return an empty table. Thanks

**Comment by Bobgy** [comment]
> I will give it some more time to see if gcloud --project=${PROJECT} container clusters list will return an empty table.

+1 to this.

It might mean your Cloud Config Connector cluster isn't properly set up, maybe it lacks some permissions or sth.

**Comment by cm185255** [comment]
`gcloud container clusters delete <name>` successfully deleted the cluster, however, `make apply` now returns a new error. 

```
# Apply management resources
kubectl --context=xx apply -f ./.build/gcp_config
unable to recognize ".build/gcp_config/compute.cnrm.cloud.google.com_v1beta1_computeaddress_xx-ip.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/compute.cnrm.cloud.google.com_v1beta1_computedisk_xx-storage-artifact-store.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/compute.cnrm.cloud.google.com_v1beta1_computedisk_xx-storage-metadata-store.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/container.cnrm.cloud.google.com_v1beta1_containercluster_xx.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-bigquery.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-cloudbuild.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-cloudsql.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-dataflow.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-dataproc.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-istio-wi.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-kubeflow-wi.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-logging.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-manages-user.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-metricwriter.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-ml.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-monitoringviewer.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-network.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-servicemanagement.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-source.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-storage.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-viewer.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-admin-workload-identity-user.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-bigquery.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-cloudbuild.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-cloudsql.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-dataflow.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-dataproc.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-logging.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-metricwriter.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-ml.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-monitoringviewer.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-source.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-storage.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-viewer.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-workload-identity-user-ml-pipeline-ui.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-workload-identity-user-ml-pipeline-visualizationserver.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-user-workload-identity-user-pipeline-runner.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-vm-logging.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-vm-policy-cloudtrace.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-vm-policy-meshtelemetry.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-vm-policy-monitoring-viewer.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-vm-policy-monitoring.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iampolicymember_xx-vm-policy-storage.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iamserviceaccount_xx-admin.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iamserviceaccount_xx-user.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/iam.cnrm.cloud.google.com_v1beta1_iamserviceaccount_xx-vm.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_anthos.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_cloudbuild.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_cloudresourcemanager.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_compute.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_container.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_gkeconnect.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_gkehub.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_iamcredentials.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_iap.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_logging.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_meshca.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_meshconfig.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_meshtelemetry.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_monitoring.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
make: *** [apply-gcp] Error 1
```

Perhaps I need to first re-create the management cluster? I'll try that first!

**Comment by cm185255** [comment]
One thing I noticed when recreating the management cluster: 

On the step `make apply-kcc`, I saw the output
```
I0825 09:04:09.005129   19943 main.go:230] reconcile iam.cnrm.cloud.google.com/IAMPolicyMember cnrm-system-MANAGED_PROJECT-owner
```

I looked in `management/instance/managed-project/iam.yaml` and found the lines 
```
metadata:
  name: cnrm-system-MANAGED_PROJECT-owner
```

Should `MANAGED_PROJECT` here be the actual name of my project?

**Comment by jlewi** [comment]
No. Its just a placeholder name for the resource and since it is applied client side by anthoscli the name doesn't really matter.

The error

```
unable to recognize ".build/gcp_config/serviceusage.cnrm.cloud.google.com_v1beta1_service_monitoring.googleapis.com.yaml": Get http://localhost:8080/api?timeout=32s: dial tcp [::1]:8080: connect: connection refused
```

Looks like a problem connecting to your kubernetes master. Why is the address localhost? That doesn't look correct. Did you delete your management cluster?

**Comment by jlewi** [comment]
@cm185255 I would suggest restricting this issue to the original issue with CNRM that @Bobgy described in the original problem as that looks like a potential issue with Cloud Config Connector that the CNRM team might be able to look into. 

For any other issues related to deploying Kubeflow lets use an issue in kubeflow/gcp-blueprints.

**Comment by cm185255** [comment]
@jlewi I apologize, the above error was encountered while running commands suggested by @Bobgy on this thread in relation to the original issue. 

Yes, when I ran the command from @Bobgy above (`gcloud container clusters delete <name> --zone <zone>`) it deleted my management cluster. After re-creating it, I am now back to the same timeout error: 

```
kubectl --context=xx wait --for=condition=Ready --timeout=600s  containercluster xx
error: timed out waiting for the condition on containerclusters/xx
make: *** [wait-gcp] Error 1 
```

The output of `kubectl --context=${CONTEXT} --namespace ${PROJECT_ID} get containercluster -o yaml` still shows for the `status` field
```
status:
    conditions:
    - lastTransitionTime: "2020-08-25T14:08:47Z"
      message: 'Update call failed: the desired mutation for the following field(s)
        is invalid: [nodeConfig.0.OauthScopes.3859019814 nodeConfig.0.MachineType
        initialNodeCount ipAllocationPolicy.# nodeConfig.0.ServiceAccount nodeConfig.0.OauthScopes.#]'
      reason: UpdateFailed
      status: "False"
      type: Ready
```
So we are right back where we started - deleting and redeploying the cluster did not resolve the issue.

**Comment by cm185255** [comment]
I found this issue https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/165 that looks quite related.

The issue emphasizes not to set `spec.initialNodeCount` in the configs. Do you know if this is being done anywhere? I have not changed any files, so it would be a default field the kubeflow package. 

I do see `initialNodeCount` set to `2` in the `spec` block of the output from `kubectl --context=${CONTEXT} --namespace ${PROJECT_ID} get containercluster -o yaml`. 
```
spec:
    clusterAutoscaling:
      autoProvisioningDefaults:
        oauthScopes:
        - https://www.googleapis.com/auth/logging.write
        - https://www.googleapis.com/auth/monitoring
        - https://www.googleapis.com/auth/devstorage.read_only
        serviceAccountRef:
          name: dip-ai-kf-vm
      enabled: true
      resourceLimits:
      - maximum: 128
        resourceType: cpu
      - maximum: 2000
        resourceType: memory
      - maximum: 16
        resourceType: nvidia-tesla-k80
    initialNodeCount: 2
    location: us-east1-b
    loggingService: logging.googleapis.com/kubernetes
    monitoringService: monitoring.googleapis.com/kubernetes
    nodeConfig:
      machineType: n1-standard-8
      metadata:
        disable-legacy-endpoints: "true"
      oauthScopes:
      - https://www.googleapis.com/auth/logging.write
      - https://www.googleapis.com/auth/monitoring
      - https://www.googleapis.com/auth/devstorage.read_only
      serviceAccountRef:
        name: dip-ai-kf-vm
      workloadMetadataConfig:
        nodeMetadata: GKE_METADATA_SERVER
    releaseChannel:
      channel: REGULAR
    workloadIdentityConfig:
      identityNamespace: kubeflow-explore-287117.svc.id.goog
```

Let me know if you have any ideas for next steps forward. To reiterate, I have not changed anything from the default configs.

**Comment by Bobgy** [comment]
> Yes, when I ran the command from @Bobgy above (gcloud container clusters delete <name> --zone <zone>) it deleted my management cluster.

Sorry, I wasn't clear enough. I asked you to delete the kubeflow cluster (corresponding to the containercluster definition), not the management cluster. Because the error you are seeing seems to indicate that cluster already exists and you cannot update some immutable fields.

Just for confirmation, did you try to deploy kubeflow into the management cluster? That's not supported. You need to use a different cluster for kubeflow.

**Comment by cm185255** [comment]
No, I am using a different cluster for kubeflow. I am following the v1.1 tutorial verbatim.

**Comment by cm185255** [comment]
My error is still the same: 
```
status:
    conditions:
    - lastTransitionTime: "2020-08-25T14:08:47Z"
      message: 'Update call failed: the desired mutation for the following field(s)
        is invalid: [nodeConfig.0.OauthScopes.3859019814 nodeConfig.0.MachineType
        initialNodeCount ipAllocationPolicy.# nodeConfig.0.ServiceAccount nodeConfig.0.OauthScopes.#]'
      reason: UpdateFailed
      status: "False"
      type: Ready
```

**Comment by Bobgy** [comment]
Thank you for your patience, looks like we need KCC team's response

**Comment by knave07** [comment]
I am also facing this issue though my message is a little bit different. 

Status of `kubectl --context=<mgmt-cntxt> --namespace <project-name> get containercluster -o yaml` is showing :

```bash
  status:
    conditions:
    - lastTransitionTime: "2020-08-25T21:00:48Z"
      message: 'Update call failed: the desired mutation for the following field(s)
        is invalid: [nodeConfig.0.MinCpuPlatform initialNodeCount]'
      reason: UpdateFailed
      status: "False"
      type: Ready
```
Environment:

- Kubeflow version: 1.1
- kfctl version: kfctl v1.0.2-0-ga476281
- Kubernetes platform: GCP
- OS: macOS 10.14.5

**Comment by xiaobaitusi** [comment]
Hi, I would suggest upgrading KCC to the latest version since it has the fix for the similar issue [#165](https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/165). Also I would suggest not using the embedded nodeConfig but instead manage the [node pools](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/crds/container_v1beta1_containernodepool.yaml) as a separate resource.

**Comment by Bobgy** [comment]
Thank you, we've upgraded KCC to 1.29.0. I think we can close this.
